File: AspNetMvcUnleashed/0672329980_source/Chapter03/CS/MvcApplication1/MvcApplication1/Controllers/NewsController.cs
using System;
using System.Collections.Generic;
using System.Web.Mvc;
using MvcApplication1.Selectors;

namespace MvcApplication1.Controllers
{
    public class NewsController : Controller
    {
        private readonly List<string> _news = new List<string>();
        private Random _rnd = new Random();

        public NewsController()
        {
            _news.Add("Moon explodes!");
            _news.Add("Stock market up 200 percent!");
            _news.Add("Talking robot created!");
        }

        public ActionResult Index()
        {
            var selectedIndex = _rnd.Next(_news.Count);
            ViewData.Model = _news[selectedIndex];
            return View();
        }


        [AjaxMethod]
        [ActionName("Index")]
        public string Index_AJAX()
        {
            var selectedIndex = _rnd.Next(_news.Count);
            return _news[selectedIndex];
        }


    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter03/VB/MvcApplication1/MvcApplication1/Controllers/NewsController.vb
Public Class NewsController
    Inherits System.Web.Mvc.Controller

    Private ReadOnly _news As New List(Of String)
    Private _rnd As New Random()

    Sub New()
        _news.Add("Moon explodes!")
        _news.Add("Stock market up 200 percent!")
        _news.Add("Talking robot created!")
    End Sub

    Function Index() As ActionResult
        Dim selectedIndex = _rnd.Next(_news.Count)
        ViewData.Model = _news(selectedIndex)
        Return View()
    End Function


    <AjaxMethod()> _
    <ActionName("Index")> _
    Function Index_AJAX() As String
        Dim selectedIndex = _rnd.Next(_news.Count)
        Return _news(selectedIndex)
    End Function


End Class
